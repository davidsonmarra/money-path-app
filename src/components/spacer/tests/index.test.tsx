import React from 'react';
import {render} from '@testing-library/react-native';
import Spacer from 'src/components/spacer';

describe('Spacer', () => {
  it('renders correctly with default props', () => {
    const {toJSON} = render(<Spacer />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders with specified width', () => {
    const {getByTestId} = render(<Spacer width={10} />);
    const spacer = getByTestId('spacer');
    expect(spacer.props.style).toEqual({width: 10, height: 0});
  });

  it('renders with specified height', () => {
    const {getByTestId} = render(<Spacer height={20} />);
    const spacer = getByTestId('spacer');
    expect(spacer.props.style).toEqual({width: 0, height: 20});
  });

  it('renders with specified width and height', () => {
    const {getByTestId} = render(<Spacer width={10} height={20} />);
    const spacer = getByTestId('spacer');
    expect(spacer.props.style).toEqual({width: 10, height: 20});
  });

  it('renders with zero width and height when props are not provided', () => {
    const {getByTestId} = render(<Spacer />);
    const spacer = getByTestId('spacer');
    expect(spacer.props.style).toEqual({width: 0, height: 0});
  });
});
