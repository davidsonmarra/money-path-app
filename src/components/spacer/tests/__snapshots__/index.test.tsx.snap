// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spacer renders correctly with default props 1`] = `
<View
  style={
    {
      "height": 0,
      "width": 0,
    }
  }
  testID="spacer"
/>
`;
